### PR TITLE
Sidenav toggle button triggers on spacebar press

### LINKED ISSUE #4559: Side navigation toggle button doesn't work on `space`
via @petermakowski comment https://github.com/canonical/juju.is/pull/431#issuecomment-1231665391

> `.p-side-navigation__toggle` element currently behaves like a link (`space` key does not trigger it on press, but that's just one example). Seems that a button element/role might be better suited here. 

**Describe the bug**

The toggle button for side navigation component is a link, not a button, and because of that you can't click it using `space` key (works only with `enter`).

It looks like a button and ideally should behave like one.

### PR DESCRIPTION
**Pull request by jmuzina**
## Done

Changes all uses of `a.p-side-navigation__toggle` and `a.p-side-navigation__toggle--in-drawer` to `<button>` elements and removed `href` attribute

Fixes #4559, [WD-11071](https://warthogs.atlassian.net/browse/WD-11071)

## QA

- Open a sidenav
    - [Accordion / Dark](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/accordion-dark)
    - [Accordion](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/accordion)
    - [Side navigation dark](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/dark)
    - [Side navigation default](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/default)
    - [Side navigation docs](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/docs)
    - [Expandable dark](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/expandable-dark)
    - [Expandable](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/expandable)
    - [Icons](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/icons)
    - [Paper](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/paper)
    - [Raw HTML](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/raw-html)
    - [Sticky](https://vanilla-framework-5084.demos.haus/docs/examples/standalone/patterns/side-navigation/sticky)
    - [Vanilla Docs](https://vanilla-framework-5084.demos.haus/docs)
- Small / Medium breakpoints:
    - Verify that the side-navigation toggle button visually appears as it does in the current production release (button styling)
    - Verify that you can open/close the side-navigation using all of the following methods:
        - Click / tap
        - Enter key
        - Space key (new functionality)
    - Verify that the open/close side-navigation button is read to you by a screen reader such as [ChromeVox](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn).
- All breakpoints:
    - Verify that the side-navigation content/structure is read to you by a screen reader such as [ChromeVox](https://chromewebstore.google.com/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn).

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots
Screenshot 2024-05-07 at 1 44 28 PM

[WD-11071]: https://warthogs.atlassian.net/browse/WD-11071?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bartaz** [comment]
I had a suspicion there was a reason why we used a link instead of the button there:

https://github.com/canonical/vanilla-framework/blob/e6d414d428e1755d99cef0b8e5cf76edf701bca5/scss/_patterns_side-navigation.scss#L120-L121

So it turns out originally we did have a button to toggle side navigation, but it was changed to links so that it can be opened just via CSS (`:target`) in https://github.com/canonical/vanilla-framework/pull/3065

So it seems we have 2 conflicting accessibility concerns here: do we want to make it accessible without JS, or do we want to make it accessible via required keys for button, which requires JS.

**Comment by bartaz** [comment]
@petermakowski Hey, any opinions on this from accessibility standpoint?

**Comment by petermakowski** [comment]
> @petermakowski Hey, any opinions on this from accessibility standpoint?

TLDR; I also think that using a button (and relying on JS) seems like the best choice.

Using a link with added JavaScript to handle spacebar on small screens (most likely a touchscreen device without a keyboard) is definitely not worth additional complexity.

The percentage of users with JavaScript disabled on small screens is likely very low. It's hard to find exact stats, but in various studies it comes at an average of 1% (but that's for all screen sizes). For mobile devices is probably even lower, the vast majority of mobile browsers have JS enabled by default.

**Comment by bartaz** [comment]
Let's remove the CSS that uses `:target` selector then, and keep it as a button.

File: scss/_patterns_side-navigation.scss [hunk 1]
```diff
@@ -167,8 +168,9 @@
     top: 0;
     z-index: 101;
 
-    .p-side-navigation:target &,
-    [class*='p-side-navigation--']:target &,
+    // :target can be used to control the sidenav with an anchor element, but this was removed in PR #5084.
+    //.p-side-navigation:target &,
+    //[class*='p-side-navigation--']:target &,
```
Comment (petermakowski) [inline scss/_patterns_side-navigation.scss:173]:
```suggestion
```

File: scss/_patterns_side-navigation.scss [hunk 2]
```diff
@@ -258,8 +260,9 @@
 
     .p-side-navigation__drawer,
     // fight specificity of expanded version
-    .p-side-navigation:target .p-side-navigation__drawer,
-    [class*='p-side-navigation--']:target .p-side-navigation__drawer,
+    // :target can be used to control the sidenav with an anchor element, but this was removed in PR #5084.
+    // .p-side-navigation:target .p-side-navigation__drawer,
+    // [class*='p-side-navigation--']:target .p-side-navigation__drawer,
```
Comment (petermakowski) [inline scss/_patterns_side-navigation.scss:265]:
```suggestion
```

File: scss/_patterns_side-navigation.scss [hunk 3]
```diff
@@ -117,8 +117,9 @@
     width: 100%;
     z-index: 102;
 
-    .p-side-navigation:target &,
-    [class*='p-side-navigation--']:target &,
+    // :target can be used to control the sidenav with an anchor element, but this was removed in PR #5084.
+    // .p-side-navigation:target &,
+    // [class*='p-side-navigation--']:target &,
```
Comment (petermakowski) [inline scss/_patterns_side-navigation.scss:122]: I'd recommend to remove commented out code, changes and relevant PRs can be tracked using git history.

```suggestion
```

**Comment by petermakowski** [comment]
I can confirm this fixes the issue I raised https://github.com/canonical/juju.is/pull/431#issuecomment-1231665391

**Comment by jmuzina** [comment]
I have a commit ready to remove those comments, just holding the push until @bartaz performs code review on his return (I'd rather not kick off a Percy build from just removing some sass comments) :)

File: scss/_patterns_side-navigation.scss (hunk 3 above)
Comment (bartaz) [reply scss/_patterns_side-navigation.scss:122]: I agree. There is no point of keeping commented code. There is a clear reason to remove that.

**Review by bartaz — APPROVED** [review]
Looks good, but let's remove the commented out code.

**Comment by jmuzina** [comment]
Removed comments & will merge after checks pass
